lexpredict_openedgar/openedgar/tasks.py

```diff
@@ -188,24 +186,29 @@ def create_filing_error(row, filing_path: str):
 
 
 @shared_task
-def process_filing_index(s3_path: str, filing_index_buffer: Union[str, bytes] = None,
+def process_filing_index(client_type: str, file_path: str, filing_index_buffer: Union[str, bytes] = None,
                          form_type_list: Iterable[str] = None, store_raw: bool = False, store_text: bool = False):
     """
     Process a filing index from an S3 path or buffer.
-    :param s3_path: S3 path to process; if filing_index_buffer is none, retrieved from here
+    :param file_path: S3 or local path to process; if filing_index_buffer is none, retrieved from here
     :param filing_index_buffer: buffer; if not present, s3_path must be set
     :param form_type_list: optional list of form type to process
     :param store_raw:
     :param store_text:
     :return:
     """
     # Log entry
-    logger.info("Processing filing index {0}...".format(s3_path))
+    logger.info("Processing filing index {0}...".format(file_path))
+
+    if client_type == "S3":
+        client = S3Client()
+    else:
+        client = LocalClient()
 
     # Retrieve buffer if not passed
     if filing_index_buffer is None:
-        logger.info("Retrieving filing index buffer from S3...")
-        filing_index_buffer = openedgar.clients.s3.get_buffer(s3_path)
+        logger.info("Retrieving filing index buffer for: {}...".format(file_path))
+        filing_index_buffer = client.get_buffer(file_path)
```

Comment:
@LRParser @mjbommar I know this is an old MR but I was trying this out today and I don't think this works correctly on 211 and other places because it is not prefixed with the DOWNLOAD_PATH.    

I think the fix is to do this ... but
```
    if client_type == "S3":
        client = S3Client()
        path_prefix = str()
    else:
        client = LocalClient()
        path_prefix = os.environ["DOWNLOAD_PATH"]

    # Retrieve buffer if not passed
    if filing_index_buffer is None:
        logger.info("Retrieving filing index buffer for: {}...".format(file_path))
        filing_index_buffer = client.get_buffer(os.path.join(path_prefix,file_path))
```
This is what is done in /openedgar/lexpredict_openedgar/openedgar/processes/edgar.py

But perhaps it would be better to have local.py set the prefix so we don't need to do it here.  There are more instances in this file where the file_path doesn't have the prefix appended.

BTW.. I love this feature as we mount an S3 bucket and with this I can just point to it.  Simpler config with all the s3 goodness.